src/NUnitEngine/nunit.engine.api/TestPackage.cs

```diff
@@ -63,8 +61,19 @@ public TestPackage(string filePath)
         /// <param name="testFiles"></param>
         public TestPackage(IList<string> testFiles)
         {
+            ID = GetNextID();
+            SubPackages = new List<TestPackage>();
+            Settings = new Dictionary<string, object>();
+
             foreach (string testFile in testFiles)
-                Add(Path.GetFullPath(testFile));
+                SubPackages.Add(new TestPackage(testFile));
+        }
+
+        private static int _nextID = 0;
+
+        private string GetNextID()
+        {
+            return unchecked(_nextID++).ToString();
```

Comment:
For my own education, why did you use unchecked here? I thought unchecked was to prevent compile time overflow checks with constant values and that "Expressions that contain non-constant terms are unchecked by default at compile time and run time."

Do we have the /checked compiler option enabled?